### PR TITLE
Fuzzing add seed, new target and remove bloat

### PR DESCRIPTION
**Pull request by 0x34d**
#### Description
Add more `go-fuzz` tests, add input seed and remove bloat.

Bug found:  https://github.com/pion/dtls/issues/591

#### Run:

```bash
go test -fuzz=FuzzUnmarshalBinary       $(pwd)
go test -fuzz=FuzzUnmarshal             $(pwd)/pkg/protocol/extension
go test -fuzz=FuzzDtlsHandshake         $(pwd)/pkg/protocol/handshake
go test -fuzz=FuzzRecordLayer           $(pwd)/pkg/protocol/recordlayer
go test -fuzz=FuzzUnpackDatagram        $(pwd)/pkg/protocol/recordlayer
```

### CODE REVIEW

**Comment by Sean-Der** [comment]
Fantastic, thank you so much @0x34d 

Can you fix the golangci-lint errors? If you don't feel comfortable doing it I can help!

**Comment by 0x34d** [comment]
@Sean-Der ping, can you rerun the workflow?

**Comment by Sean-Der** [comment]
@0x34d you are in the org now so you can run yourself :)

sorry for slowing you down

**Comment by 0x34d** [comment]
ping for merge @hasheddan @stv0g @Sean-Der

**Comment by stv0g** [comment]
I think it would be nicer to put the fuzz corpus into files. The fuzzer reada them automatically from `testdata/TestName`

**Comment by 0x34d** [comment]
> I think it would be nicer to put the fuzz corpus into files. The fuzzer reada them automatically from `testdata/TestName`

Done https://github.com/pion/dtls/pull/592/commits/816e08482150cc42b54c5988a158049cc73937af and leave others as they are.

File: fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// SPDX-License-Identifier: MIT
+package dtls
+
+import (
+	"os"
+	"testing"
+)
+
+func FuzzUnmarshalBinary(f *testing.F) {
+	TestResumeClient, err := os.ReadFile("testdata/seed/TestResumeClient.raw")
```
Comment (stv0g) [inline fuzz_test.go:11]: Please have a look at the [Go fuzzing tutorial](https://go.dev/doc/tutorial/fuzz)

The Go fuzzing enging has built-in support for reading files from the `testdata` directory.

Comment (0x34d) [reply fuzz_test.go:11]: Oh, My bad https://go.dev/security/fuzz/#glos-seed-corpus
```txt
and the files in the testdata/fuzz/{FuzzTestName} directory within the package
```

Comment (0x34d) [reply fuzz_test.go]: > Oh, My bad https://go.dev/security/fuzz/#glos-seed-corpus
> 
> ```
> and the files in the testdata/fuzz/{FuzzTestName} directory within the package
> ```

For a minute, I thought I was wrong.
Anything within `testdata/fuzz/{FuzzTestName}` that crashes cannot be accepted as a seed for fuzzing. So, yes, my previous `commit`  was on point.

**Comment by 0x34d** [comment]
ping

**Comment by Sean-Der** [comment]
Merged with https://github.com/pion/dtls/commit/d7964378674c5a64b6a7f9222f1ff6a8209885dc

Thank you so much @0x34d sorry this didn't happen quicker

**Pull request closed without merging**